litestar/contrib/sqlalchemy/repository/_async.py

```diff
@@ -605,15 +635,18 @@ async def _list_and_count_basic(
         Returns:
             Count of records returned by query using 2 queries, ignoring pagination.
         """
-        statement = statement if statement is not None else self.statement
+        statement = self._to_lambda_stmt(statement)
         statement = self._apply_filters(*filters, statement=statement)
         statement = self._filter_select_by_kwargs(statement, kwargs)
-        count_statement = statement.with_only_columns(
-            sql_func.count(self.get_id_attribute_value(self.model_type)),
-            maintain_column_froms=True,
-        ).order_by(None)
+
+        def count_statement(statement: StatementLambdaElement) -> StatementLambdaElement:
```

Comment:
This should be a method instead of an inline function sicne it does not capture anything from the outer functions scope

Reply:
Take a look at the refactoring when you have a moment.  I've also modified the `delete_many` and `update_many` methods to use lambdas.